src/app/chat/view.nim

```diff
@@ -43,7 +44,9 @@ QtObject:
 
   proc setActiveChannelByIndex*(self: ChatsView, index: int) {.slot.} =
     if(self.chats.chats.len == 0): return
-
+    var response = self.status.chat.markAllChannelMessagesRead(self.activeChannel.id)
+    if not response.hasKey("error"):
+      self.chats.clearUnreadMessagesCount(self.activeChannel.chatItem)
```

Comment:
Probably want to revisit this. I think it's better to leverage nim's built-in error handling than relying on a `JsonNode` type return from the Status API